notebooks/official/feature_store/offline_feature_serving_from_bigquery_with_feature_registry.ipynb

```diff
@@ -0,0 +1,472 @@
+        "\n",
+        "FEATURE_GROUP_ID = \"fhfv_fg_unique\"  # @param {type:\"string\"}\n",
+        "\n",
+        "DATA_SOURCE = \"gs://cloud-samples-data-us-central1/vertex-ai/feature-store/datasets/movie_prediction.csv\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "cd580a0679ce"
+      },
+      "source": [
+        "### Write BigQuery table for use in Feature Registry"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1ff4481243a8"
+      },
+      "outputs": [],
+      "source": [
+        "bigframes.options.bigquery.project = PROJECT_ID\n",
+        "bigframes.options.bigquery.location = LOCATION\n",
+        "df = bigframes.pandas.read_csv(DATA_SOURCE)\n",
+        "df[\"timestamp\"] = pd.to_datetime(df[\"timestamp\"], utc=True)\n",
+        "df = df.rename(columns={\"timestamp\": \"feature_timestamp\"})\n",
+        "df.to_gbq(BQ_TABLE_URI, if_exists=\"replace\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "818a36b9da86"
+      },
+      "source": [
+        "### Create Feature Registry Features"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1f915ddd4669"
+      },
+      "outputs": [],
+      "source": [
+        "fg: FeatureGroup = FeatureGroup.create(\n",
+        "    f\"{FEATURE_GROUP_ID}\",\n",
+        "    fs_utils.FeatureGroupBigQuerySource(\n",
+        "        uri=f\"bq://{BQ_TABLE_URI}\", entity_id_columns=[\"users\"]\n",
+        "    ),\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0dba1c02883c"
+      },
+      "outputs": [],
+      "source": [
+        "movie_feature: Feature = fg.create_feature(\"movies\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c71964219962"
+      },
+      "source": [
+        "### Fetch with same entity at multiple timestamps"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8fb9774db748"
+      },
+      "outputs": [],
+      "source": [
+        "entity_df = pd.DataFrame(\n",
+        "    data={\n",
+        "        \"users\": [\"alice\", \"alice\"],\n",
+        "        \"timestamp\": [\n",
+        "            pd.Timestamp(\"2021-09-14T09:36\"),\n",
+        "            pd.Timestamp(\"2023-12-12T13:13\"),\n",
+        "        ],\n",
+        "    },\n",
+        ")\n",
+        "\n",
+        "offline_store.fetch_historical_feature_values(\n",
+        "    entity_df=entity_df,\n",
+        "    features=[movie_feature],\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "f9ecae3005df"
+      },
+      "source": [
+        "### Fetch with different entities"
```

Comment:
Suggested edit for H3: "Fetch historical feature values for multiple entities"

Add a description here, similar to the preceding H3: "Fetch feature values for multiple entities and timestamps. <Mention whether both the timestamps are considered for each entity.>"